### PR TITLE
Bump Ktlint to 1.0.1

### PR DESCRIPTION
**Pull request by Goooler**
Follow up #1808.

https://github.com/pinterest/ktlint/releases/tag/1.0.1

### CODE REVIEW

**Review by nedtwigg — COMMENTED** [review]
Sorry, forgot to click "Submit review". I think this PR can be much much smaller, let me know if I'm wrong.

File: lib/src/main/java/com/diffplug/spotless/kotlin/KtLintStep.java [hunk 1]
```diff
@@ -36,7 +36,7 @@ public class KtLintStep {
 	// prevent direct instantiation
 	private KtLintStep() {}
 
-	private static final String DEFAULT_VERSION = "1.0.0";
```
[the remark this answers is not in the document]
Comment (Goooler) [reply lib/src/main/java/com/diffplug/spotless/kotlin/KtLintStep.java:39]: You are right, update for now. Can we keep these test cases?

[the remark this answers is not in the document]
Comment (nedtwigg) [reply lib/src/main/java/com/diffplug/spotless/kotlin/KtLintStep.java:39]: Yes, that would be great!

**Comment by nedtwigg** [comment]
Released in `plugin-gradle 6.23.0` and `plugin-maven 2.41.0`.

**Comment by Goooler** [comment]
`userData` is unused for now, you should try `editorConfigOverride`, check https://github.com/diffplug/spotless/tree/main/plugin-gradle#ktlint

**Pull request merged**